content/en/docs/reference/node/seccomp.md

```diff
@@ -0,0 +1,117 @@
+---
+content_type: reference
+title: Seccomp and Kubernetes
+weight: 80
+---
+
+<!-- overview -->
+
+Seccomp stands for secure computing mode and has been a feature of the Linux
+kernel since version 2.6.12. It can be used to sandbox the privileges of a
+process, restricting the calls it is able to make from userspace into the
+kernel. Kubernetes lets you automatically apply seccomp profiles loaded onto a
+{{< glossary_tooltip text="node" term_id="node" >}} to your Pods and containers.
+
+## Seccomp fields
+
+{{< feature-state for_k8s_version="v1.19" state="stable" >}}
+
+There are three ways to specify a seccomp profile for a
+{{< glossary_tooltip text="pod" term_id="pod" >}}:
+
+- for the whole Pod using [`spec.securityContext.seccompProfile`](/docs/reference/kubernetes-api/workload-resources/pod-v1/#security-context)
+- for a single container using [`spec.containers[*].securityContext.seccompProfile`](/docs/reference/kubernetes-api/workload-resources/pod-v1/#security-context-1)
+- for an [ephermal container](/docs/concepts/workloads/pods/ephemeral-containers) using [`spec.ephemeralContainers[*].securityContext.seccompProfile`](/docs/reference/kubernetes-api/workload-resources/pod-v1/#security-context-2)
+
+{{% code_sample file="pods/security/seccomp/fields.yaml" %}}
+
+The Pod in the example above runs as `Unconfined`, while the
+`ephemeral-container` specifically defines `RuntimeDefault`. If the ephemeral
+container would not have set the `securityContext.seccompProfile` field
+explicitly, then the value would be inherited by the Pod. The same applies to
+the container, which runs a `Localhost` profile `my-profile.json`.
+
+Generally speaking, fields from (ephemeral) containers have a higher priority
+than the Pod level value, while containers which do not set the seccomp field
+are being inherited by the Pod.
+
+{{< note >}}
+It is not possible to apply a seccomp profile to a Pod or container running with
+`privileged: true` set in the container's `securityContext`. Privileged
+containers always run as `Unconfined`.
+{{< /note >}}
+
+The following values are possible for the `seccompProfile.type`:
+
+`Unconfined`
+: The workload runs without any seccomp restrictions.
+
+`RuntimeDefault`
+: A default seccomp profile defined by the {{< glossary_tooltip text="container runtime" term_id="container-runtime" >}} is applied.
+
+`Localhost`
+: The `localhostProfile` will be applied, which has to be available on the node disk (on Linux it's `/var/lib/kubelet/seccomp`).
+
+### `Localhost` profiles
```

Comment:
I never thought of it, but reviewing docs I started wondering. Can init container of a Pod be used to install the seccomp profile for the main container? Or profiles existence will be validated before Pod start executing init containers?

Reply:
Yes, that's possible. For example with:

```yaml
apiVersion: v1
kind: Pod
metadata:
  name: pod
spec:
  volumes:
    - name: seccomp-profiles
      hostPath:
        path: /var/lib/kubelet/seccomp
  initContainers:
    - name: init
      image: docker.io/library/debian:stable
      command: ["bash", "-c", 'echo {\"defaultAction\":\"SCMP_ACT_LOG\"} > /seccomp/profile.json']
      volumeMounts:
        - name: seccomp-profiles
          mountPath: /seccomp
  containers:
    - name: container
      securityContext:
        seccompProfile:
          type: Localhost
          localhostProfile: profile.json
      image: docker.io/library/debian:stable
      args: ["sleep", "infinity"]
```

Reply:
Can you pls add a note that the existance of a file is calculated before the pod start, not on admission? Just to clarify that this (using init container to configure security) is actually possible

Reply:
Do you mean container start? If the existence were checked before Pod start, this would have to happen between binding to a node and the kubelet beginning to perform Pod creation, so an init container couldn't help.

Reply:
My only point here is that we need to tell in documentation that the validation of seccomp profile existence is checked when container is created, not at Pod admission time. Thus the trick where Init container conifigures seccomp for the main container can work.

Reply:
Also the expected behavior  when the configured seccomp profile is not present - would be an interesting thing to highlight in the docs.

Reply:
I am fine to do it as a follow up if this is easier

Reply:
I added a note to the localhost section above (https://github.com/kubernetes/website/pull/47967/files#diff-814041cc7d665890cb57edc7c13098f2b658bdc2119988e98b13a27731f7863bR61-R67)